raytracing/drawing.py

```diff
@@ -0,0 +1,324 @@
+import matplotlib.pyplot as plt
+from matplotlib import patches, transforms
+from matplotlib import text as mplText
+import matplotlib.path as mpath
+from typing import List
+import numpy as np
+import math
+
+
+class Drawing:
+    """ The drawing of any element.
+
+    A Drawing can be a composition of different drawing components (ex.: A lens drawing is two arrows).
+
+    Args:
+        *components: The required drawing components (of type `matplotlib.patches.Patch`) that define the Drawing.
+
+            These drawing patches should be instantiated at x = 0 to allow for proper positioning and scaling.
+
+    Examples:
+        Create a Drawing from multiple patches
+
+        >>> arrowUp = ArrowPatch(dy=5)
+        >>> arrowDown = ArrowPatch(dy=-5)
+        >>> lensDrawing = Drawing([arrowUp, arrowDown])
+
+        Apply the Drawing on a figure at x=10
+
+        >>> fig, axes = plt.subplots()
+        >>> lensDrawing.applyTo(axes, x=10)
+
+        Take advantage of the auto-scaling feature by updating the Drawing after the figure's limits have changed
+        (on a zoom callback)
+
+        >>> lensDrawing.update()
+
+        Update the Drawing's position
+
+        >>> lensDrawing.update(x=5)
+
+    """
+
+    def __init__(self, components: List[patches.Patch], label: str = None,
+                 x=0, y=0, fixedWidth=False):
+        self.components = components
+        self.label = None
+
+        self.axes = None
+        self.x = x
+        self.y = y
+        self.useAutoScale = not fixedWidth
+
+        if label is not None:
+            self.label = Label(text=label, x=self.centroid[0], y=self.halfHeight() * 1.2)
+
+    @property
+    def hasLabel(self) -> bool:
+        if self.label is None:
+            return False
+        return True
+
+    def applyTo(self, axes: plt.Axes, x: float = None, y: float = None):
```

Comment:
`Drawing` should be GUI-agnostic: I would not use matplotlib terminology because we could change it later. You could define your own (x,y) coordinates, which would be in dataspace anyway(?) so regardless of the choice of graphics backend, you could use your class.